src/component/plotly-graph/parse.js

```diff
@@ -108,4 +112,99 @@ function parseDim (result, opts, dim) {
   }
 }
 
+function willFigureHang (result) {
+  const data = result.figure.data
+
+  // cap the number of traces
+  if (data.length > 200) {
+    return true
+  }
+
+  const numberOfPtsPerTraceType = {}
+
+  for (let i = 0; i < data.length; i++) {
+    const trace = data[i] || {}
+    const type = trace.type || 'scatter'
+
+    if (!(type in numberOfPtsPerTraceType)) {
+      numberOfPtsPerTraceType[type] = 0
+    }
+
+    numberOfPtsPerTraceType[type] += estimateDataLength(trace)
+
+    // cap the number of point per trace type
+    if (numberOfPtsPerTraceType[type] > maxPtsPerTraceType(type)) {
+      return true
+    }
+
+    // other special cases
+
+    // mesh3d will `alphahull` and 1000+ pts
+    if (
+      type === 'mesh3d' &&
+      'alphahull' in trace && Number(trace.alphahull) >= 0 &&
+      findMaxArrayLength(trace) > 1000
+    ) {
+      return true
+    }
+  }
+}
+
+// Consider the array of maximum length as a proxy to determine
+// the number of points to be drawn. In general, this estimate
+// can be (much) smaller than the true number of points plotted
+// when it does not match the length of the other coordinate arrays.
+function findMaxArrayLength (cont) {
+  const lengths = Object.keys(cont)
+    .filter(k => Array.isArray(cont[k]))
+    .map(k => cont[k].length)
+
+  return Math.max(...lengths)
+}
+
+function estimateDataLength (trace) {
+  // special case for e.g. parcoords and splom traces
+  if (Array.isArray(trace.dimensions)) {
+    return trace.dimensions
+      .map(findMaxArrayLength)
+      .reduce((a, v) => a + v)
+  }
+
+  return findMaxArrayLength(trace)
+}
+
+function maxPtsPerTraceType (type) {
+  switch (type) {
+    case 'scattergl':
+    case 'splom':
+    case 'pointcloud':
+      return 1e7
+
+    case 'scatterpolargl':
+    case 'heatmap':
+    case 'heatmapgl':
+      return 1e6
+
+    case 'scatter3d':
+    case 'surface':
+    case 'mesh3d':
+      return 5e5
+
+    case 'parcoords':
+      return 5e5
+    case 'scattermapbox':
+      return 5e5
+
+    case 'histogram':
+    case 'histogram2d':
+    case 'histogram2dcontour':
+    case 'box':
+    case 'violin':
+      return 1e5
```

Comment:
Can these really only handle 1e5? They're not actually displaying every point, I'd have thought we could bump up to 1e6 perhaps.

Reply:
I was thinking of `box` and `violin` for (box)points turned on. We could make a special case for those I guess.

Reply:
Ah good call - most of the time outliers are a tiny fraction of total points, but not always...